test/test_config.py

```diff
@@ -1,138 +1,412 @@
+import os
+import shutil
+from tempfile import mkdtemp
+
 from test.picardtestcase import PicardTestCase
 
-from picard import config
+from picard.config import (
+    BoolOption,
+    Config,
+    FloatOption,
+    IntListOption,
+    IntOption,
+    ListOption,
+    Option,
+    TextOption,
+)
+
+class TestPicardConfigCommon(PicardTestCase):
+
+    def setUp(self):
+        super().setUp()
+        self.tmp_directory = mkdtemp()
+        self.configpath = os.path.join(self.tmp_directory, 'test.ini')
+        self.config = Config.from_file(None, self.configpath)
+        self.config.application["version"] = "testing"
+        logging.disable(logging.ERROR)
+        Option.registry = {}
+
+    def tearDown(self):
+        shutil.rmtree(self.tmp_directory)
+
+
+class TestPicardConfig(TestPicardConfigCommon):
+
+    def test_remove(self):
+        TextOption("setting", "text_option", "abc")
+
+        self.config.setting["text_option"] = "def"
+        self.assertEqual(self.config.setting["text_option"], "def")
+
+        self.config.setting.remove("text_option")
+        self.assertEqual(self.config.setting["text_option"], "abc")
+
+
+class TestPicardConfigTextOption(TestPicardConfigCommon):
+
+    ### TextOption
+    def test_text_opt_convert(self):
+        opt = TextOption("setting", "text_option", "abc")
+        self.assertEqual(opt.convert(123), "123")
+
+    def test_text_opt_no_config(self):
+        TextOption("setting", "text_option", "abc")
+
+        # test default, nothing in config yet
+        self.assertEqual(self.config.setting["text_option"], "abc")
+        self.assertEqual(type(self.config.setting["text_option"]), str)
+
+    def test_text_opt_set_read_back(self):
+        TextOption("setting", "text_option", "abc")
+
+        # set option to "def", and read back
+        self.config.setting["text_option"] = "def"
+        self.assertEqual(self.config.setting["text_option"], "def")
+        self.assertEqual(type(self.config.setting["text_option"]), str)
+
+    def test_text_opt_set_None(self):
+        TextOption("setting", "text_option", "abc")
+
+        # set option to None
+        self.config.setting["text_option"] = None
+        self.assertEqual(self.config.setting["text_option"], "")
+
+    def test_text_opt_set_empty(self):
+        TextOption("setting", "text_option", "abc")
+
+        # set option to ""
+        self.config.setting["text_option"] = ""
+        self.assertEqual(self.config.setting["text_option"], "")
+
+    def test_text_opt_xxx(self):
+        TextOption("setting", "text_option", "abc")
+
+        # store invalid value in config file directly
+        self.config.setValue('setting/text_option', object)
+        self.assertEqual(self.config.setting["text_option"], 'abc')
+
+
+class TestPicardConfigBoolOption(TestPicardConfigCommon):
+
+    ### BoolOption
+    def test_bool_opt_convert(self):
+        opt = BoolOption("setting", "bool_option", False)
+        self.assertEqual(opt.convert(1), True)
```

Comment:
You can use `self.assertTrue(value)` or `self.assertFalse(value)` for these boolean checks.

Reply:
Those aren't equivalent:

```
value = "a"
self.assertTrue(value)        # <---- pass
self.assertEqual(value, True)  # <---- not pass
```

https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertTrue says:
> This method should also be **avoided** when more specific methods are available (e.g. assertEqual(a, b) instead of assertTrue(a == b)), because they provide a better error message in case of failure.